internal/web_server/web/handlers/api/organization_admin_api.go

```diff
@@ -0,0 +1,83 @@
+package api
+
+import (
+	"net/http"
+
+	"github.com/Molnes/Nyhetsjeger/internal/config"
+	"github.com/Molnes/Nyhetsjeger/internal/models/sessions"
+	"github.com/Molnes/Nyhetsjeger/internal/models/users"
+	"github.com/Molnes/Nyhetsjeger/internal/models/users/access_control"
+	"github.com/Molnes/Nyhetsjeger/internal/utils"
+	"github.com/Molnes/Nyhetsjeger/internal/web_server/web/views/components/dashboard_components/access_settings_components"
+	"github.com/labstack/echo/v4"
+)
+
+type OrganizationAdminApiHandler struct {
+	sharedData *config.SharedData
+}
+
+// Creates a new OrganizationAdminApiHandler
+func NewOrganizationAdminApiHandler(sharedData *config.SharedData) *OrganizationAdminApiHandler {
+	return &OrganizationAdminApiHandler{sharedData}
+}
+
+// Registers the organization admin related handlers to the given echo group
+func (oah *OrganizationAdminApiHandler) RegisterOrganizationAdminHandlers(g *echo.Group) {
+	g.POST("/access-control/admin", oah.postAddAdminByEmail)
+	g.DELETE("/access-control/admin", oah.deleteAdminByEmail)
+}
+
+// Handles a post request to add an admin by email. Email expected in form data.
+func (oah *OrganizationAdminApiHandler) postAddAdminByEmail(c echo.Context) error {
+	email := c.FormValue("email")
+	if email == "" {
+		return echo.NewHTTPError(http.StatusBadRequest, "Missing email")
+	}
+
+	session, err := oah.sharedData.SessionStore.Get(c.Request(), sessions.SESSION_NAME)
+	if err != nil {
+		return err
+	}
+	caller := session.Values[sessions.USER_DATA_VALUE].(users.UserSessionData)
+	if caller.Email == email {
+		return echo.NewHTTPError(http.StatusBadRequest, "Cannot modify own admin status")
+	}
+
+	userAdmin, err := access_control.AddAdmin(oah.sharedData.DB, email)
+	if err != nil {
+		if err == access_control.ErrEmailAlreadyAdmin {
+			return echo.NewHTTPError(http.StatusConflict, "Email is already an admin")
+		}
+		return err
+	}
+
+	return utils.Render(c, http.StatusCreated, access_settings_components.AdminTableRow(userAdmin))
+}
+
+// expects email in json body
+func (oah *OrganizationAdminApiHandler) deleteAdminByEmail(c echo.Context) error {
+	email := c.QueryParam("email")
```

Comment:
is this JSON?

Reply:
bb2d5e17a5955e420674b42b04c938fc04f0c27c